src/software/ai/navigator/path_planner/theta_star_path_planner_test.cpp

```diff
@@ -135,26 +145,31 @@ TEST(TestThetaStarPathPlanner, test_theta_star_path_planner_single_obstacle_alon
     std::vector<Obstacle> obstacles = {
         Obstacle(Polygon({Point(1, 1), Point(2, 1), Point(2, -1), Point(1, -1)}))};
 
-    std::unique_ptr<PathPlanner> planner =
-        std::make_unique<ThetaStarPathPlanner>(field, obstacles);
-
+    std::unique_ptr<PathPlanner> planner = std::make_unique<ThetaStarPathPlanner>();
+    std::vector<Point> path_points;
 
-    auto path_points = planner->findPath(start, dest);
-
-    // We expect to find a path
-    ASSERT_TRUE(path_points);
+    try
+    {
+        path_points = std::get<std::vector<Point>>(
+            planner->findPath(start, dest, field, obstacles));
+    }
+    catch (const std::bad_variant_access&)
+    {
+        // not a valid path
+        EXPECT_TRUE(0);
+    }
 
     // The path should start at exactly the start point and end at exactly the dest
-    EXPECT_EQ(start, path_points->front());
-    EXPECT_EQ(dest, path_points->back());
+    EXPECT_EQ(start, path_points.front());
+    EXPECT_EQ(dest, path_points.back());
 
     // Make sure the path does not exceed a bounding box
     Rectangle bounding_box({-0.1, 1.2}, {3.1, -1.2});
-    checkPathDoesNotExceedBoundingBox(*path_points, bounding_box);
+    checkPathDoesNotExceedBoundingBox(path_points, bounding_box);
 
     // Can't make sure the path does not go through any obstacles
     // since start is blocked
-    // checkPathDoesNotIntersectObstacle(*path_points, obstacles);
+    // checkPathDoesNotIntersectObstacle(path_points, obstacles);
```

Comment:
why is this commented out?

Reply:
"Can't make sure the path does not go through any obstacles since start is blocked"-> it's a bad test that will fail if enabled, but it's also a weird case that should be tested in a different way

Reply:
Wow sorry, should have read more carefully. Could we remove the commented out code, but leave the comment as:
```
    // NOTE: Can't make sure the path does not go through any obstacles
    // since start is blocked.
```

Reply:
Done